ui/packages/catalog-ui-search/src/main/webapp/react-component/container/map-context-menu/index.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import CopyCoordinates from '../../presentation/copy-coordinates'
+import { Menu, MenuItem } from '../../menu'
+import styled from '../../styles/styled-components'
+
+const Icon = styled.div`
+  display: inline-block;
+  text-align: center;
+  width: ${props => props.theme.minimumFontSize};
+`
+const Title = styled.div`
+  display: inline-block;
+  margin-left: ${props => props.theme.minimumSpacing};
+  margin-right: ${props => props.theme.minimumSpacing};
+`
+const Description = styled.div`
+  display: block;
+  margin-left: calc(
+    ${props => props.theme.minimumSpacing} +
+      ${props => props.theme.minimumFontSize}
+  );
+  margin-right: ${props => props.theme.minimumSpacing};
+  line-height: normal;
+  margin-top: calc(0 - ${props => props.theme.minimumSpacing});
+  margin-bottom: ${props => props.theme.minimumSpacing};
+`
+
+interface Props {
+  onChange: (value: string) => void
+  mouseLat: number | null | undefined
+  mouseLon: number | null | undefined
+  target: string | null | undefined
+  clickDms: string
+  clickLat: string
+  clickLon: string
+  clickMgrs: string
+  clickUtmUps: string
+  selectionCount: number
+  closeMenu: () => void
+}
+
+export const MapContextMenu = (props: Props) => {
+  const hasTarget = typeof props.target === 'string'
+  const hasSelection = props.selectionCount > 0
+  const hasMouseCoordinates =
+    typeof props.mouseLat !== 'undefined' &&
+    typeof props.mouseLon !== 'undefined'
+  const menuItems = []
+  if (hasMouseCoordinates) {
```

Comment:
❓ Can we pull up these predicate blocks into separate functions outside of the component. I think it would make it a little easier to read.

Reply:
Let me know if you think this new version is better.